Modules/_csv.c

```diff
@@ -1327,12 +1255,23 @@ csv_writerow(PyObject *op, PyObject *seq)
         return NULL;
     }
 
-    /* Join all fields in internal buffer.
-     */
-    join_reset(self);
+    writer = PyUnicodeWriter_Create(0);
+    if (writer == NULL) {
+        Py_DECREF(iter);
+        return NULL;
+    }
+
     while ((field = PyIter_Next(iter))) {
-        int append_ok;
         int quoted;
+        PyObject *str_field = NULL;
+        bool is_none = (field == Py_None);
```

Comment:
`is_none` doesn't say what *is* refers to so just keep the previous variable name, this is really what a None field means.